examples/simple-test/main.py

```diff
@@ -0,0 +1,99 @@
+from pathlib import Path
+
+import fire
+import pyds
+from sparrow_datums import FrameAugmentedBoxes, PType, non_max_suppression
+
+import sparrow_mlpipes as mlp
+
+WIDTH = 1920
+HEIGHT = 1080
+FPS = 20
+
+
+def print_objects_pad(_, info, __):
+    # Intiallizing object counter with 0.
+    batch_meta = mlp.get_batch_meta(info)
+    pyds.nvds_acquire_meta_lock(batch_meta)
+    for frame in mlp.pyds_generator(batch_meta.frame_meta_list):
+        frame_meta = mlp.get_frame_meta(frame)
+        print(frame_meta.frame_num)
+        for user in mlp.pyds_generator(frame_meta.frame_user_meta_list):
+            user_meta = mlp.get_user_meta(user)
+            augmented_boxes = mlp.get_output_tensor(user_meta, 0)
+            boxes = non_max_suppression(
+                FrameAugmentedBoxes(
+                    augmented_boxes,
+                    ptype=PType.relative_tlbr,
+                    image_width=WIDTH,
+                    image_height=HEIGHT,
+                )
+            )
+            print(boxes.shape)
+            for box in boxes:
+                obj_meta = pyds.nvds_acquire_obj_meta_from_pool(batch_meta)
+                obj_meta.rect_params.left = WIDTH * box.x1
+                obj_meta.rect_params.top = HEIGHT * box.y1
+                obj_meta.rect_params.width = WIDTH * box.w
+                obj_meta.rect_params.height = HEIGHT * box.h
+                obj_meta.rect_params.border_width = 3
+                obj_meta.rect_params.border_color.set(0, box.label / 91, box.score, 0)
+                pyds.nvds_add_obj_meta_to_frame(frame_meta, obj_meta, None)
+
+    pyds.nvds_release_meta_lock(batch_meta)
+    return mlp.Gst.PadProbeReturn.OK
+
+
+def main(
+    path: str,
+    config_path: str = "./nvinfer.config",
+) -> None:
+    """Get multifilesrc working."""
+    pipeline = mlp.Gst.Pipeline()
+    multifilesrc = mlp.make_element(
+        "multifilesrc", location=Path(path, "%d.jpeg"), start_index=0
```

Comment:
I like this way of structuring it. Does gstreamer complain about the path object?

Reply:
Nope - no issues. I’m not sure all 20 twenty “test” images are getting read in though. Looking now.

Reply:
Running `python main.py /code/data/datasets/images/test/` seems to be at least loading all 20 of the "test" images once I got everything pointed to the correct directory. The output from `print(frame_meta.frame_num)` is something like:
...
19
(0, 6)
20
(0, 6)